lib/fabrik/VirtualHostRelationMapper.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+const Promise = require('bluebird');
+const logger = require('../logger');
+const bosh = require('../bosh');
+const cloudController = require('../cf').cloudController;
+const virtualHostStore = require('../iaas').virtualHostStore;
+
+class VirtualHostRelationMapper {
+  constructor() {
+    this.cache = {};
+    this.director = bosh.director;
+    this.cloudController = cloudController;
+    this.virtualHostStore = virtualHostStore;
+  }
+
+  static get instanceConstructor() {
+    return VirtualHostRelationMapper;
+  }
+
+  createVirtualHostRelation(deploymentName, instanceId) {
+    logger.info(`Storing deployment name : '${deploymentName}' for virtual_host instance id : '${instanceId}'`);
+    this.cache[instanceId] = deploymentName;
+    const data = {
+      'deployment_name': deploymentName,
+      'instance_guid': instanceId
+    };
+    return this.virtualHostStore.putFile(data);
+  }
+
+  getDeploymentName(instanceId) {
+    logger.info(`Fetching deployment name for virtual_host instance id : '${instanceId}'`);
+    if (this.cache[instanceId] === undefined) {
+      logger.info(`Cache miss for deployment name for virtual_host instance id ${instanceId}, Will load it from store.`);
+      return this.loadCacheforInstance(instanceId);
+    }
+    return Promise.resolve(this.cache[instanceId]);
```

Comment:
nothing wrong with this, but as a convention in such scenarios the method is wrapped with Promise.try & a normal return is performed. Look at other code references and for keeping it consistent, please change it.